internal/service/alarms/internal/middleware.go

```diff
@@ -0,0 +1,55 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"time"
+
+	"github.com/getkin/kin-openapi/openapi3filter"
+	oapimiddleware "github.com/oapi-codegen/nethttp-middleware"
+	"github.com/openshift-kni/oran-o2ims/internal/service/alarms/api/generated"
+)
+
+type Middleware = func(http.Handler) http.Handler
+
+// LogDuration log time taken to complete a request.
+// TODO: This is just get started with middleware but should be replaced with something that's more suitable for production i.e OpenTelemetry
+// https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/examples/prometheus/main.go
+func LogDuration() Middleware {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			startTime := time.Now()
+			next.ServeHTTP(w, r)
+			slog.Info(fmt.Sprintf("%s took %s", r.RequestURI, time.Since(startTime)))
+		})
+	}
+}
+
+// AlarmsOapiValidation to validate all incoming requests as specified in the spec
+func AlarmsOapiValidation() Middleware {
+	// This also validates the spec file
+	swagger, err := generated.GetSwagger()
+	if err != nil {
+		// TODO: handle this error
+		slog.Error(fmt.Sprintf("failed to get swagger: %s", err))
+	}
+
+	// Clear out the servers array in the swagger spec, that skips validating
+	// that server names match. We don't know how this thing will be run.
+	swagger.Servers = nil
+
+	return oapimiddleware.OapiRequestValidatorWithOptions(swagger, &oapimiddleware.Options{
+		Options: openapi3filter.Options{
+			AuthenticationFunc: openapi3filter.NoopAuthenticationFunc, // No auth needed even when we have something in spec
+		},
+		ErrorHandler: func(w http.ResponseWriter, message string, statusCode int) {
```

Comment:
Shouldn't we write a generic error handler function like this and here create getOranReqErrFunc and getOranRespErrFunc?

Reply:
Sure moved it into func